### PR TITLE
ENT-1200

### PR DESCRIPTION
**Pull request by muhammad-ammar**
https://openedx.atlassian.net/browse/ENT-1200

These changes will add support for POST request on search/all endpoint.

NOTE: We have disabled the DRF browsable api interface for search/all endpoint only for now because it is causing issues for GET request after new changes.

### CODE REVIEW

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 1]
```diff
@@ -361,6 +361,21 @@ def test_results_include_aggregation_key(self):
...
+    path = reverse('api:v1:search-all-list')
+
+    def test_post(self):
+        """
+        Verify that POST request works as expected for `AggregateSearchViewSet`
+        """
+        course = CourseFactory(key='course:edX+DemoX', title='ABCs of Ͳҽʂէìղց')
+        data = {'content_type': 'course', 'aggregation_key': ['course:edX+DemoX']}
+        expected = {'previous': None, 'results': [], 'next': None, 'count': 0}
+        response = self.client.post(self.path, data=data)
```
Comment (muhammad-ammar) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py]: @georgebabey I have created the course but data is not returned on both GET and POST even though data is indexed. But if send a GET request without query params or POST request without data then it returns the data. Any idea what I am missing here?

File: course_discovery/apps/api/v1/views/search.py [hunk 2]
```diff
@@ -92,6 +94,32 @@ def filter_facet_queryset(self, queryset):
...
+            if isinstance(value, list):
+                request_filters.setlist(param, value)
+            else:
+                request_filters[param] = value
+
+        return request_filters
+
+
+class CatalogDataViewSet(viewsets.GenericViewSet):
+    renderer_classes = [renderers.JSONRenderer]
```
Comment (muhammad-ammar) [inline course_discovery/apps/api/v1/views/search.py:115]: Not to Reviewers: This will disable the browsable api for `search/all` endpoint.

**Comment by muhammad-ammar** [comment]
@georgebabey @zubair-arbi @mushtaqak This is ready for review.

File: course_discovery/apps/api/v1/views/search.py [hunk 3]
```diff
@@ -92,6 +94,32 @@ def filter_facet_queryset(self, queryset):
         return queryset
 
 
+class CatalogDataFilterBackend(HaystackFilter):
+
+    @staticmethod
+    def get_request_filters(request):
+        if request.method == 'GET':
+            return HaystackFilter.get_request_filters(request)
```
Comment (zubair-arbi) [inline course_discovery/apps/api/v1/views/search.py:102]: Can we use super here `return super(CatalogDataFilterBackend, self).get_request_filters(request)`?

**Review by zubair-arbi — COMMENTED** [review]
I am also not sure how should we resolve searching on GET request.

File: course_discovery/apps/api/v1/views/search.py (hunk 3 above)
Comment (muhammad-ammar) [reply course_discovery/apps/api/v1/views/search.py:102]: This is a static method, `self` is not available here.

File: course_discovery/apps/api/v1/views/search.py [hunk 4]
```diff
@@ -92,6 +94,44 @@ def filter_facet_queryset(self, queryset):
...
+        """Disable the form for POST"""
+
+        if method == 'POST':
+            return False
+
+        return super(BrowsableAPIRendererWithoutPost, self).show_form_for_method(view, method, request, obj)
+
+
+class CatalogDataViewSet(viewsets.GenericViewSet):
+    renderer_classes = [BrowsableAPIRendererWithoutPost, renderers.JSONRenderer]
```
Comment (muhammad-ammar) [inline course_discovery/apps/api/v1/views/search.py]: @zubair-arbi @georgebabey I have made more changes and now the browsable interface works as before but will not show the **form for POST request**.

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍

**Review by brittneyexline — APPROVED** [review]

**Pull request merged**